examples/chart/teleport-kube-agent/templates/deployment.yaml

```diff
@@ -26,6 +26,14 @@ spec:
         {{- if .Values.insecureSkipProxyTLSVerify }}
         - "--insecure"
         {{- end }}
+        securityContext:
+          allowPrivilegeEscalation: false
+          capabilities:
+            drop:
+            - all
+          readOnlyRootFilesystem: true
```

Comment:
does this affect mounted volumes?
we definitely need the `/var/lib/teleport` volume writable.
on your cluster with PSP enforced, can you exec into the pod and check that `/var/lib/teleport` gets populated?

Reply:
> does this affect mounted volumes?
No


readOnlyRootFs can be thought of as setting all the files included in the container as read-only. The shim layer that docker normally creates for writes is read-only and as such binaries or other files shouldn't be able to be modified. Any volume mounts to the mount namespace follow their own parameters for whether they are readonly or not. AFAICT the teleport deployment already sets /var/lib/teleport as an emptydir mount, so that would be writable. 

https://github.com/gravitational/teleport/blob/master/examples/chart/teleport-kube-agent/templates/deployment.yaml#L54-L55
https://github.com/gravitational/teleport/blob/master/examples/chart/teleport-kube-agent/templates/deployment.yaml#L63-L64